### PR TITLE
Write PGP keys to publish-conf when doing publish setup

### PR DESCRIPTION
**Pull request by MaciejG604**
Make `publish setup` write down PGP keys to publish-conf
Always check if keys are uploaded to a keyserver

### CODE REVIEW

File: modules/cli/src/main/scala/scala/cli/commands/publish/PublishSetup.scala [hunk 1]
```diff
@@ -259,28 +259,32 @@ object PublishSetup extends ScalaCommand[PublishSetupOptions] {
...
-        os.write.over(dest, updatedContent)
-        logger.message("") // printing an empty line, for readability
-        logger.message(s"Wrote ${CommandUtils.printablePath(dest)}")
-        written = written :+ dest
+        if (extraLines.nonEmpty) {
+          val updatedContent = currentContent ++
+            extraLines.toArray.flatMap(_.getBytes(StandardCharsets.UTF_8))
+          os.write.over(dest, updatedContent)
+          logger.message("") // printing an empty line, for readability
+          logger.message(s"Wrote ${CommandUtils.printablePath(dest)}")
```
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/commands/publish/PublishSetup.scala]: NIT
```suggestion
                logger.message( // printing an empty line for readability
        s"""
           |Wrote ${CommandUtils.printablePath(dest)}""".stripMargin
      )
```

File: modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala [hunk 2]
```diff
@@ -240,4 +251,223 @@ class PublishSetupTests extends ScalaCliSuite {
...
+        "--public-key",
+        "file:key.pub",
+        projDir
+      ).call(
+        cwd = root,
+        mergeErrIntoOut = true,
+        env = envs
+      )
+
+      System.err.write(res.out.bytes)
```
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala]: ?

File: modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala [hunk 3]
```diff
@@ -240,4 +251,223 @@ class PublishSetupTests extends ScalaCliSuite {
...
+        "--secret-key",
+        "file:key.skr",
+        projDir
+      ).call(
+        cwd = root,
+        mergeErrIntoOut = true,
+        env = envs
+      )
+
+      System.err.write(res.out.bytes)
```
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala]: ?

File: modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala [hunk 4]
```diff
@@ -240,4 +251,223 @@ class PublishSetupTests extends ScalaCliSuite {
...
+        "--public-key",
+        "file:key.pub",
+        projDir
+      ).call(
+        cwd = root,
+        mergeErrIntoOut = true,
+        env = envs
+      )
+
+      System.err.write(res2.out.bytes)
```
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala:425]: ?

File: modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala [hunk 5]
```diff
@@ -240,4 +251,223 @@ class PublishSetupTests extends ScalaCliSuite {
...
+        "--secret-key",
+        "value:whatever",
+        projDir
+      ).call(
+        cwd = root,
+        mergeErrIntoOut = true,
+        env = envs
+      )
+
+      System.err.write(res.out.bytes)
```
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala]: ?

File: modules/cli/src/main/scala/scala/cli/commands/publish/checks/PgpSecretKeyCheck.scala [hunk 6]
```diff
@@ -66,228 +88,344 @@ final case class PgpSecretKeyCheck(
...
-                            (if (options.publishParams.setupCi)
-                               " Alternatively, pass --random-secret-key"
-                             else "")
-                      )
-                    )
-                  }
-              }
+  private lazy val keyServers: Either[BuildException, Seq[Uri]] = {
+    val rawKeyServers = options.sharedPgp.keyServer.filter(_.trim.nonEmpty)
+    if (rawKeyServers.filter(_.trim.nonEmpty).isEmpty)
```
Comment (lwronski) [inline modules/cli/src/main/scala/scala/cli/commands/publish/checks/PgpSecretKeyCheck.scala]: You don't need to run `trim` twice in `rawKeyServers` variable

File: modules/cli/src/main/scala/scala/cli/commands/publish/checks/PgpSecretKeyCheck.scala [hunk 7]
```diff
@@ -66,228 +88,344 @@ final case class PgpSecretKeyCheck(
...
+              case Right(Right(_)) => true
+              case Right(Left(msg)) =>
+                logger.debug(
+                  s"""Response from $keyServer:
+                     |$msg
+                     |""".stripMargin
+                )
+                false
+              case Left(err) =>
+                logger.debug(s"Error checking $keyId at $keyServer: $err")
```
Comment (lwronski) [inline modules/cli/src/main/scala/scala/cli/commands/publish/checks/PgpSecretKeyCheck.scala]: If it is an error message, maybe we should always print it.

```suggestion
                logger.error(s"Error checking $keyId at $keyServer: $err")
```

**Review by lwronski — APPROVED** [review]

**Pull request merged**